protocols/randhound/struct.go

```diff
@@ -0,0 +1,137 @@
+package randhound
+
+import (
+	"sync"
+	"time"
+
+	"github.com/dedis/cothority/crypto"
+	"github.com/dedis/cothority/sda"
+	"github.com/dedis/crypto/abstract"
+)
+
+// RandHound is the main protocol struct and implements the
+// sda.ProtocolInstance interface.
+type RandHound struct {
+	*sda.TreeNodeInstance
+
+	mutex sync.Mutex
+
+	// Session information
+	nodes   int       // Total number of nodes (client + servers)
+	groups  int       // Number of groups
+	faulty  int       // Maximum number of Byzantine servers
+	purpose string    // Purpose of protocol run
+	time    time.Time // Timestamp of initiation
+	cliRand []byte    // Client-chosen randomness (for initial sharding)
+	sid     []byte    // Session identifier
+
+	// Group information
+	server              [][]*sda.TreeNode  // Grouped servers
+	group               [][]int            // Grouped server indices
+	threshold           []int              // Group thresholds
```

Comment:
I am usually sticking to the singular, so that when you iterate over the variable you have `threshold[i]` instead of `thresholds[i]` which is more intuitive to me. In the above example the same holds for `server[i][j]` etc. I should probably check if there are some Golang coding conventions on that.